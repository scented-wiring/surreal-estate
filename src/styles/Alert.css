.alert-success {
  color: green;
}

.alert-error {
  color: red;
}
