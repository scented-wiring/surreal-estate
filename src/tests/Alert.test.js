import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

it("displays an error message", () => {
  const { getByText, asFragment } = render(<Alert message="Error!" />);

  expect(getByText(/Error/).textContent).toBe("Error!");
  expect(getByText(/Error/)).toHaveClass("alert-error");
  expect(asFragment()).toMatchSnapshot();
});

it("displays a success message", () => {
  const { getByText, asFragment } = render(
    <Alert message="Success!!!!" success />
  );

  expect(getByText(/Success/).textContent).toBe("Success!!!!");
  expect(getByText(/Success/)).toHaveClass("alert-success");
  expect(asFragment()).toMatchSnapshot();
});

it("does not render an error or success message if message props is empty", () => {
  const { asFragment } = render(<Alert message="" />);
  const alert = asFragment();

  expect(alert).toMatchSnapshot();
});
