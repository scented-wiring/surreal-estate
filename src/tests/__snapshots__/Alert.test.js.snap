// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert displays a success message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-success"
  >
    Success!!!!
  </div>
</DocumentFragment>
`;

exports[`Alert displays an error message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert does not render an error or success message if message props is empty 1`] = `<DocumentFragment />`;
